PythonRegression/tests/features/steps/api_test_steps.py

```diff
@@ -104,16 +197,112 @@ def check_neighbors_post_removal(step):
+    ports = [port1,port2]
+    
+    api1 = prepare_api_call(node1,machine)
+    api2 = prepare_api_call(node2,machine)
+        
+    response1 = api1.get_neighbors()
+    response2 = api2.get_neighbors()
+    neighbors1 = list(response1['neighbors'])
+    neighbors2 = list(response2['neighbors'])
+    host = hosts[0]
+    port = ports[0]
+    address1 = "udp://" + str(host) + ":" + port     
+    host = hosts[1]
+    port = ports[1]
+    address2 = "udp://" + str(host) + ":" + port 
+    
+    logger.info(address1)
+    logger.info(address2)
+    logger.info("Checking if nodes are paired")
+    
+    containsNeighbor = False
+    for neighbor in range(len(neighbors1)):
+        if neighbors1[neighbor]['address']:
+            containsNeighbor = True
+            logger.info("Neighbor found")
+
+    
+    if containsNeighbor == False:
+        api1.add_neighbors([address2.decode()])
+        api2.add_neighbors([address1.decode()])
+        logger.info("Nodes paired")
+    
+        
+    containsNeighbor = False
+    for neighbor in range(len(neighbors2)):
+        if neighbors2[neighbor]['address']:
+            containsNeighbor = True 
+            logger.info("Neighbor found")
+
+    if containsNeighbor == False:
+        api2.add_neighbors([address1.decode()])
+        api1.add_neighbors([address2.decode()]) 
+        logger.info("Nodes paired")
+        
+        
+    response = api1.get_neighbors()
+    logger.info(response)
+    response = api2.get_neighbors()
+    logger.info(response)
+        
+     
+@step(r'a transaction with the tag "([^"]*)" is sent from "([^"]*)"')
+def send_transaction(step,tag,nodeName):
+    logger.info('Preparing Transaction...')
+    logger.info('Node: %s',nodeName)
+    machine = config['machine']
+    config['tag'] = tag
+    api = prepare_api_call(nodeName,machine)  
+    txn = \
+        ProposedTransaction(
+            address = 
+            Address(testAddress),
+            
+            message = TryteString.from_unicode('Test Transaction propagation'),
+            tag = Tag(tag),
+            value = 0,
+            )
     
+    logger.info("Sending Transaction...")
+    txn_sent = api.send_transfer(depth=3, transfers=[txn])
+    logger.info("Giving the transaction time to propagate...")
+    sleep(10)
+   
+   
+@step(r'findTransaction is called with the same tag on "([^"]*)"')
+def find_transaction_is_called(step,nodeName):
+    logger.info(nodeName)
+    api = prepare_api_call(nodeName, config['machine'])    
+    logger.info("Searching for Transaction")
+    response = api.find_transactions(tags=[config['tag']])    
+    config['findTransactionResponse'] = response
     
-      
-  
-  
-                                
+@step(r'the Transaction should be found')
+def check_transaction_response(step):
+    logger.info("Checking response...")
+    response = config['findTransactionResponse']
+    assert len(response['hashes']) != 0, 'Transaction not found'
+    logger.info("Response found")  
```

Comment:
maybe print the repsonse?

Reply:
Printing the response was part of the default activity and it clutters the shell print if you are looking for transactions with a tag/address that is associated with a large number of transactions

Reply:
Print how many transactions were found